parsers/FO.py

```diff
@@ -44,48 +54,128 @@ def fetch_production(
             f"This parser cannot retrieve data for zone {zone_key!r}",
         )
 
-    if target_datetime is not None:
-        # There is a API endpoint at https://www.sev.fo/api/elproduction/last7days
-        # but it's currently returning nothing at all. Last checked on 2022-08-09
-        raise ParserException(
-            PARSER, "This parser is not yet able to parse past dates", zone_key=zone_key
-        )
+    now = datetime.now(timezone.utc)
+    target_datetime = (
+        now if target_datetime is None else target_datetime.astimezone(timezone.utc)
+    )
+
+    # API provides max one day of backlog
+    target_date_utc_to = datetime.combine(target_datetime, time(), tzinfo=timezone.utc)
+    target_date_utc_from = target_date_utc_to - timedelta(days=1)
+
+    # API takes local days as query parameters and returns data for the corresponding interval as UTC
+    target_date_local_to = target_date_utc_to.astimezone(tz=TIMEZONE)
+    target_date_local_from = target_date_utc_from.astimezone(tz=TIMEZONE)
+
+    # during SDT local time is UTC + 0 so day boundaries align
+    # but during DST local time is UTC + 1 so the data for UTC 23:00 is at 00:00 of the next local day, which will thus need to be requested
+    if target_date_local_to.dst():
+        target_date_local_to = target_date_local_to + timedelta(days=1)
 
+    url = f"https://www.sev.fo/api/elproduction/search?type=item&group=powerplant&from={target_date_local_from.strftime('%Y%m%d')}170000&to={target_date_local_to.strftime('%Y%m%d')}170000"
     ses = session or Session()
-    url = "https://www.sev.fo/api/realtimemap/now"
     response: Response = ses.get(url)
-    obj = response.json()
+    if not response.ok:
+        raise ParserException(
+            PARSER,
+            f"Exception when fetching production error code: {response.status_code}: {response.text}",
+            zone_key,
+        )
+
+    hourly_production_for_each_power_plant = response.json()
 
     production_breakdown_list = ProductionBreakdownList(logger)
-    production_mix = ProductionMix()
-    for key, value in obj.items():
-        if "Sum" in key or "Test" in key or "VnVand" in key:
-            # "VnVand" is the sum of hydro (Mýrarnar + Fossá + Heygar)
+
+    hourly_production_for_each_power_plant_by_hour = groupby(
+        hourly_production_for_each_power_plant, key=lambda x: x["date"]
```

Comment:
An `itemgetter` is both faster and easier read/understand than a lambda IMO, so let's use that.